File: programmers/Lv.1/완주하지_못한_선수.py
# 완주하지 못한 선수
# https://school.programmers.co.kr/learn/courses/30/lessons/42576

from collections import Counter

def solution(participant, completion):
    answer = list(Counter(participant) - Counter(completion))
    return ''.join(answer)

print(solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]))

File: programmers/고득점Kit/해시/완주하지_못한_선수.js
function solution(participant, completion) {
  let runner = new Map();

  for(let i = 0; i < participant.length; i++) {
    let a = participant[i];
    let b = completion[i];
    runner.set(a, (runner.get(a) || 0) + 1);
    runner.set(b, (runner.get(b) || 0) - 1);
  }

  for (let [key, value] of runner) {
    if(value > 0) return key;
  }

  return;
}

console.log(solution(["leo", "kiki", "eden"],	["eden", "kiki"]))
console.log(solution(["marina", "josipa", "nikola", "vinko", "filipa"],["josipa", "filipa", "marina", "nikola"]))
console.log(solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]))